pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	for _, property := range integrationProperties {
+		name := property.Name
+		value := property.Value
+		switch name {
+		case "ENABLED", "COMMENT":
+			// We set this using the SHOW INTEGRATION call so let's ignore it here
 		case "NETWORK_POLICY":
-			if err := d.Set("network_policy", v.(string)); err != nil {
-				return fmt.Errorf("unable to set network policy for security integration")
+			if err := d.Set("network_policy", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SYNC_PASSWORD":
+			if err := d.Set("sync_password", helpers.StringToBool(value)); err != nil {
+				return diag.FromErr(err)
 			}
 		case "RUN_AS_ROLE":
-			if err := d.Set("provisioner_role", v.(string)); err != nil {
-				return fmt.Errorf("unable to set provisioner role for security integration")
+			if err := d.Set("run_as_role", value); err != nil {
+				return diag.FromErr(err)
 			}
 		default:
-			log.Printf("[WARN] unexpected security integration property %v returned from Snowflake", k)
+			log.Printf("[WARN] unexpected property %v returned from Snowflake", name)
 		}
 	}
 
-	return err
+	return nil
 }
 
-// UpdateSCIMIntegration implements schema.UpdateFunc.
-func UpdateSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// UpdateContextSCIMIntegration implements schema.UpdateFunc.
+func UpdateContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	id := d.Id()
-
-	stmt := snowflake.NewSCIMIntegrationBuilder(id).Alter()
-
-	var runSetStatement bool
-
-	if d.HasChange("scim_client") {
-		runSetStatement = true
-		stmt.SetString(`SCIM_CLIENT`, d.Get("scim_client").(string))
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	var runSet, runUnset bool
+	set, unset := sdk.NewScimIntegrationSetRequest(), sdk.NewScimIntegrationUnsetRequest()
+	if d.HasChange("enabled") {
+		runSet = true
+		set.WithEnabled(d.Get("enabled").(bool))
 	}
-
-	if d.HasChange("provisioner_role") {
-		runSetStatement = true
-		stmt.SetString(`RUN_AS_ROLE`, d.Get("provisioner_role").(string))
-	}
-
-	// We need to UNSET this if we remove all api blocked prefixes.
 	if d.HasChange("network_policy") {
-		v := d.Get("network_policy").(string)
-		if len(v) == 0 {
-			if err := snowflake.Exec(db, fmt.Sprintf(`ALTER SECURITY INTEGRATION %v UNSET NETWORK_POLICY`, id)); err != nil {
-				return fmt.Errorf("error unsetting network_policy")
-			}
+		networkPolicyID := sdk.NewAccountObjectIdentifier(d.Get("network_policy").(string))
+		if networkPolicyID.Name() != "" {
+			runSet = true
+			set.WithNetworkPolicy(networkPolicyID)
 		} else {
-			runSetStatement = true
-			stmt.SetString(`NETWORK_POLICY`, d.Get("network_policy").(string))
+			runUnset = true
+			unset.WithNetworkPolicy(true)
 		}
 	}
+	if d.GetRawConfig().AsValueMap()["sync_password"].IsNull() {
```

Comment:
Why RawConfig? I suspect this may be connected to the fact the `sync_password` change is not being detected on update "normally" (It could be resolved by writing customdiff to "refresh" the value).